ksql-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenSpec.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.codegen;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ListMultimap;
+import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.util.GenericRowValueTypeEnforcer;
+import io.confluent.ksql.function.udf.Kudf;
+import io.confluent.ksql.name.FunctionName;
+import io.confluent.ksql.schema.ksql.ColumnRef;
+import io.confluent.ksql.util.KsqlException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+public class CodeGenSpec {
```

Comment:
Would this benefit from being split into a builder and an immutable spec?

Reply:
Oh I like that - I'll have a follow up PR as soon as I merge this one so I can get this in asap